app/gui2/src/components/lexical/index.ts

```diff
@@ -15,9 +16,36 @@ export interface LexicalPlugin {
   register: (editor: LexicalEditor) => void
 }
 
+export function lexicalTheme(theme: Record<string, string>): EditorThemeClasses {
+  function getTheme(className: string) {
+    if (!(className in theme))
+      console.warn(`Referenced class ${className} not found in lexical theme.`, theme)
+    return theme[className]!
+  }
+  return {
+    text: {
+      strikethrough: getTheme('strikethrough'),
+      italic: getTheme('italic'),
+      bold: getTheme('bold'),
+    },
+    quote: getTheme('quote'),
```

Comment:
I'm wondering if it would be possible to somehow skip the indirection here, and use the css module as a theme object directly. That way the theme could be fully defined by the css file alone.

Reply:
Yeah, that would be nice. The theme is hierarchical but CSS modules are flat, so I encoded the hierarchy in the class names.